### PR TITLE
feat: implemented identity->salesforce mappings

### PR DESCRIPTION
**Pull request by davidkelley**

### CODE REVIEW

File: src/routes/customer-registration/events.post.js [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+    return value
+  }, 'external id presence check')
+  .label('CustomerEvent')
+  .unknown(true)
+
+const ResponseSchema = Joi.object({
+  message: Joi.string().required(),
+  salesforceRequest: Joi.object().required().unknown(true),
+  salesforceResponse: Joi.object().unknown(true)
```
[the remark this answers is not in the document]
Comment (davidkelley) [reply src/routes/customer-registration/events.post.js]: Updated ResponseSchema to make salesforceResponse optional and added a test covering the 202 (Salesforce disabled) path.

File: src/lib/salesforce/mappers/address.js [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+  if (validFrom && validFrom.getTime() > now) {
+    return false
+  }
+
+  if (validTo && validTo.getTime() <= now) {
+    return false
+  }
+
+  return true
+}
```
[the remark this answers is not in the document]
Comment (davidkelley) [reply src/lib/salesforce/mappers/address.js]: Swapped the validTo comparison to a strict less-than and added an address mapper test to cover the boundary case.

File: src/lib/salesforce/mappers/contact.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    method: 'PATCH',
+    url: `/services/data/${apiVersion}/sobjects/Contact/APHA_DefraCustomerId__c/${encodeURIComponent(
+      contact.contactid
+    )}`,
+    referenceId: 'ContactUpsert',
+    body
+  }
+}
+
+export { buildContactRequest }
```
[the remark this answers is not in the document]
Comment (davidkelley) [reply src/lib/salesforce/mappers/contact.js]: Added contact mapper unit tests covering inline address fallback, dual email mapping, account reference wiring, and the empty-payload error path.

File: src/lib/salesforce/client.js [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+
+    cancel()
+
+    const body = await this.parseResponseBody(response)
+
+    return { response, body }
+  }
+}
+
+export const salesforceClient = new SalesforceClient()
```
[the remark this answers is not in the document]
Comment (davidkelley) [reply src/lib/salesforce/client.js]: Added salesforce/client.test.js to cover token caching/expiry, concurrent refresh locking, timeout handling, composite failures, and missing-credential errors.

File: src/lib/salesforce/mappers/address.js [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+    }) ||
+      valueWithFlag(contact, 'defra_addrbuildingnumber', {
+        allowMissingFlag: true
+      }),
+    valueWithFlag(contact, 'defra_addrstreet', { allowMissingFlag: true }),
+    valueWithFlag(contact, 'defra_addrlocality', { allowMissingFlag: true }),
+    valueWithFlag(contact, 'defra_addrdependentlocality', {
+      allowMissingFlag: true
+    })
+  ])
```
[the remark this answers is not in the document]
Comment (davidkelley) [reply src/lib/salesforce/mappers/address.js]: Adjusted address mapping to consider building numbers/premises consistently across address detail and inline contact paths, plus tests to cover both.

File: src/routes/customer-registration/events.post.js [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+      isBadRequest
+        ? error.message
+        : 'Failed to process customer event for Salesforce',
+      [
+        new HTTPError(
+          isBadRequest ? 'VALIDATION_ERROR' : 'UNKNOWN',
+          error.message
+        )
+      ]
+    )
```
[the remark this answers is not in the document]
Comment (davidkelley) [reply src/routes/customer-registration/events.post.js]: Non-mapping failures now emit specific codes (TIMEOUT or SALESFORCE_API_ERROR); MappingErrors still use VALIDATION_ERROR.

File: src/lib/salesforce/mappers/address.js [hunk 7]
```diff
@@ -0,0 +1,361 @@
...
+    body[countryLabelField] = address.countryLabel
+  }
+}
+
+export {
+  selectAddress,
+  mapInlineContactAddress,
+  mapRegisteredAddress,
+  applyAddressToBody
+}
```
[the remark this answers is not in the document]
Comment (davidkelley) [reply src/lib/salesforce/mappers/address.js]: Added address mapper tests for valid-from/to scoring, building number/premises handling, inline contact addresses, and applyAddressToBody.
